### PR TITLE
[C#] feat: Azure AI Search data source

### PR DESCRIPTION
**Pull request by singhk97**
## Linked issues

closes: #1410 (issue number)

## Details

* Azure AI Search data store 
* Indexer project to create/delete the required index to test the bot.

## Attestation Checklist

- [x] My code follows the style guidelines of this project

- I have checked for/fixed spelling, linting, and other errors
- I have commented my code for clarity
- I have made corresponding changes to the documentation (updating the doc strings in the code is sufficient)
- My changes generate no new warnings
- I have added tests that validates my changes, and provides sufficient test coverage. I have tested with:
  - Local testing
  - E2E testing in Teams
- New and existing unit tests pass locally with my changes

### CODE REVIEW

File: dotnet/samples/08.datasource.azureaisearch/AzureAISearchBot/AzureAISearchDataSource.cs [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+            foreach (string field in selectedFields)
+            {
+                options.Select.Add(field);
+            }
+            SearchResults<Restaurant> search = SearchClient.Search<Restaurant>(query, options);
+
+            //// VECTOR SEARCH ////
+            //// Get the restaurants with description that is the most similar to the user input.
+            /*SearchOptions options = new();
```
Comment (corinagum) [inline dotnet/samples/08.datasource.azureaisearch/AzureAISearchBot/AzureAISearchDataSource.cs:62]:
```suggestion
            /* SearchOptions options = new();
```

File: .gitignore [hunk 2]
```diff
@@ -122,4 +122,19 @@ env/.env.local
...
 /dotnet/samples/NuGet.Config
-/dotnet/samples/LocalPkg/
\ No newline at end of file
+/dotnet/samples/LocalPkg/
+/dotnet/samples/08.datasource.azureaisearch/bin/Debug/net6.0
+/dotnet/samples/08.datasource.azureaisearch/obj
+/dotnet/samples/08.datasource.azureaisearch/.vs
+/dotnet/samples/08.datasource.azureaisearch/AzureAISearchBot/bin/Debug
+/dotnet/samples/08.datasource.azureaisearch/AzureAISearchBot/obj
+/dotnet/samples/08.datasource.azureaisearch/AzureAISearchBot/AzureAISearchBot.csproj.user
```
Comment (corinagum) [inline .gitignore:131]: Do these really need to go in the root .gitignore?

File: dotnet/samples/08.datasource.azureaisearch/AzureAISearchBot/AzureAISearchDataSource.cs [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+            if (string.IsNullOrEmpty(query))
+            {
+                return new RenderedPromptSection<string>("");
+            }
+
+            List<string> selectedFields = new() { "RestaurantId", "RestaurantName", "Description", "Category", "Tags", "Rating", "Address" };
+            List<string> searchFields = new() { "Category", "Tags", "RestaurantName" };
+
+            //// TEXT SEARCH ////
+            //// Get the restaurants with the query has high lexical relevance to category, tags, and restaurant name.
```
Comment (corinagum) [inline dotnet/samples/08.datasource.azureaisearch/AzureAISearchBot/AzureAISearchDataSource.cs:47]:
```suggestion
            //// Get the restaurants with the query that has high lexical relevance to category, tags, and restaurant name.
```

File: dotnet/samples/08.datasource.azureaisearch/AzureAISearchBot/AzureAISearchDataSource.cs [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+            */
+
+            //// HYBRID SEARCH ////
+            //// Search using both vector and text search
+            /*SearchOptions options = new();
+            ReadOnlyMemory<float> vectorizedQuery = await this._GetEmbeddingVector(query);
+            foreach (string field in searchFields)
+            {
+                options.SearchFields.Add(field);
+            }
```
Comment (corinagum) [inline dotnet/samples/08.datasource.azureaisearch/AzureAISearchBot/AzureAISearchDataSource.cs:82]: Is this all supposed to be commented out?

**Review by corinagum — APPROVED** [review]

File: dotnet/samples/08.datasource.azureaisearch/AzureAISearchBot/AzureAISearchDataSource.cs (hunk 4 above)
Comment (singhk97) [reply dotnet/samples/08.datasource.azureaisearch/AzureAISearchBot/AzureAISearchDataSource.cs:82]: yes users have 3 search type to choose from "simple", "vector" & "hybrid". Default is "simple"

File: .gitignore (hunk 2 above)
Comment (singhk97) [reply .gitignore:131]: fixed

**Review by corinagum — APPROVED** [review]

**Pull request merged**